lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -31,7 +31,52 @@ using namespace mlir::torch::Torch;
 // vector.
 template <typename OpTy>
 static LogicalResult
-checkAndGetPoolingParameters(OpTy op, ConversionPatternRewriter &rewriter,
+checkAndGetPooling1dParameters(OpTy op, ConversionPatternRewriter &rewriter,
```

Comment:
This function is almost identical to the 2d case. I think it should work for both 1d and 2d if you change

```
SmallVector<Value, 1> kernelSizeTorchInt;
```

to

```
SmallVector<Value> kernelSizeTorchInt;
```

Reply:
Thank you! Fixed